storage/src/test/java/org/apache/kafka/server/log/remote/metadata/storage/RemoteLogMetadataManagerTestUtils.java

```diff
@@ -0,0 +1,111 @@
+    static final long METADATA_TOPIC_RETENTION_MS = 24 * 60 * 60 * 1000L;
+
+    static Builder builder() {
+        return new Builder();
+    }
+
+    static class Builder {
+        private String bootstrapServers;
+        private boolean startConsumerThread;
+        private RemoteLogMetadataTopicPartitioner remoteLogMetadataTopicPartitioner;
+        private Map<String, Object> overrideRemoteLogMetadataManagerProps = Collections.emptyMap();
+
+        private Builder() {
+        }
+
+        public Builder bootstrapServers(String bootstrapServers) {
+            this.bootstrapServers = Objects.requireNonNull(bootstrapServers);
+            return this;
+        }
+
+        public Builder startConsumerThread(boolean startConsumerThread) {
+            this.startConsumerThread = startConsumerThread;
+            return this;
+        }
+
+        public Builder remoteLogMetadataTopicPartitioner(RemoteLogMetadataTopicPartitioner remoteLogMetadataTopicPartitioner) {
+            this.remoteLogMetadataTopicPartitioner = Objects.requireNonNull(remoteLogMetadataTopicPartitioner);
+            return this;
+        }
+
+        public Builder overrideRemoteLogMetadataManagerProps(Map<String, Object> overrideRemoteLogMetadataManagerProps) {
+            this.overrideRemoteLogMetadataManagerProps = Objects.requireNonNull(overrideRemoteLogMetadataManagerProps);
+            return this;
+        }
+
+        public TopicBasedRemoteLogMetadataManager build() {
+            String logDir = TestUtils.tempDirectory("rlmm_segs_").getAbsolutePath();
+            TopicBasedRemoteLogMetadataManager topicBasedRemoteLogMetadataManager = new TopicBasedRemoteLogMetadataManager(startConsumerThread);
+
+            // Initialize TopicBasedRemoteLogMetadataManager.
+            Map<String, Object> configs = new HashMap<>();
+            configs.put(REMOTE_LOG_METADATA_COMMON_CLIENT_PREFIX + CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+            configs.put(BROKER_ID, 0);
+            configs.put(LOG_DIR, logDir);
+            configs.put(REMOTE_LOG_METADATA_TOPIC_PARTITIONS_PROP, METADATA_TOPIC_PARTITIONS_COUNT);
+            configs.put(REMOTE_LOG_METADATA_TOPIC_REPLICATION_FACTOR_PROP, METADATA_TOPIC_REPLICATION_FACTOR);
+            configs.put(REMOTE_LOG_METADATA_TOPIC_RETENTION_MS_PROP, METADATA_TOPIC_RETENTION_MS);
+
+            log.debug("TopicBasedRemoteLogMetadataManager configs before adding overridden properties: {}", configs);
+            // Add override properties.
+            configs.putAll(overrideRemoteLogMetadataManagerProps);
+            log.debug("TopicBasedRemoteLogMetadataManager configs after adding overridden properties: {}", configs);
+
+            topicBasedRemoteLogMetadataManager.configure(configs);
+            if (remoteLogMetadataTopicPartitioner != null)
+                topicBasedRemoteLogMetadataManager.setRlmTopicPartitioner(remoteLogMetadataTopicPartitioner);
```

Comment:
This is duplicated with `TopicBasedRemoteLogMetadataManagerHarness#initializeRemoteLogMetadataManager`, maybe we should extract them as a helper method?